xds/internal/client/singleton.go

```diff
@@ -141,3 +143,49 @@ func NewWithConfigForTesting(config *bootstrap.Config, watchExpiryTimeout time.D
 	}
 	return &Client{clientImpl: cl, refCount: 1}, nil
 }
+
+// NewClientWithBootstrapContents returns an xds client for this config,
+// separate from the global singleton.  This should be used for testing
+// purposes only.
+func NewClientWithBootstrapContents(contents []byte) (*Client, error) {
```

Comment:
Make it explicit with `NewClientWithBootstrapContentsForTesting`?

Reply:
I think this will be needed for production when we do federation.